Tests/PubNubTests/Mocking/MockURLSession.swift

```diff
@@ -71,60 +71,144 @@ class MockURLSessionTask: URLSessionDataTask {
     actualSession.sessionDescription = sessionDescription
   }
+  
+  convenience init(
+    tasks: [URLSessionTask],
+    configuration: URLSessionConfiguration = .ephemeral,
+    delegate: URLSessionDelegate? = FileSessionManager(),
+    delegateQueue: OperationQueue? = .main
+  ) {
+    self.init(configuration: configuration, delegate: delegate, delegateQueue: delegateQueue)
+    self.state.lockedWrite { $0.tasks = tasks }
+  }
 
   var dataDelegate: URLSessionDataDelegate? {
-    return urlSessionEvents as? URLSessionDataDelegate
+    return delegate as? URLSessionDataDelegate
   }
 
   var sessionDelegate: HTTPSessionDelegate? {
-    return urlSessionEvents as? HTTPSessionDelegate
+    return delegate as? HTTPSessionDelegate
   }
 
-  func resume(task: MockURLSessionTask) {
-    delegateQueue.addOperation {
-      let taskIndex = self.state.lockedRead { $0.tasks.firstIndex(of: task) ?? 0 }
+  func resume(task: MockURLSessionDataTask) {
+    delegateQueue.addOperation { [weak self] in
+      guard let strongSelf = self else {
+        print("Failed to resumed due to weak self")
+        return
+      }
+  
+      let taskIndex = strongSelf.state.lockedRead { $0.tasks.firstIndex(of: task) ?? 0 }
 
-      if let mockTask = self.responseForTask?(task, taskIndex) {
+      if let mockTask = strongSelf.responseForDataTask?(task, taskIndex) {
         mockTask.mockState = .completed
         if let data = mockTask.mockData {
-          self.dataDelegate?.urlSession?(self.actualSession, dataTask: task, didReceive: data)
+          strongSelf.dataDelegate?.urlSession?(strongSelf.actualSession, dataTask: task, didReceive: data)
         }
-        self.dataDelegate?.urlSession?(self.actualSession, task: mockTask, didCompleteWithError: mockTask.error)
+        strongSelf.dataDelegate?.urlSession?(strongSelf.actualSession, task: mockTask, didCompleteWithError: mockTask.error)
       }
     }
   }
+  
+  func resume(task: MockURLSessionUploadTask) {
+    delegateQueue.addOperation { [weak self] in
+      guard let strongSelf = self else {
+        print("Failed to resumed due to weak self")
+        return
+      }
+      
+      if let data = task.mockData {
+        (strongSelf.delegate as? URLSessionDataDelegate)?.urlSession?(strongSelf.actualSession, dataTask: task, didReceive: data)
+      }
+      
+      (strongSelf.delegate as? URLSessionDataDelegate)?.urlSession?(strongSelf.actualSession, task: task, didCompleteWithError: task.error)
+    }
+  }
+  
+  func resume(task: MockURLSessionDownloadTask) {
+    delegateQueue.addOperation { [weak self] in
+      guard let strongSelf = self else {
+        print("Failed to resumed due to weak self")
+        return
+      }
+      
+      if let location = task.mockDownloadLocation {
+        (strongSelf.delegate as? URLSessionDownloadDelegate)?.urlSession(strongSelf.actualSession, downloadTask: task, didFinishDownloadingTo: location)
```

Comment:
Codacy found an issue: [Line should be 120 characters or less: currently 152 characters](https://app.codacy.com/manual/PubNub/swift/pullRequest?prid=6302911)